mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java

```diff
@@ -381,4 +400,65 @@ public boolean isFound() {
             return message.get();
         }
     }
+
+    public CompletableFuture<Stream<MessageIdAttachmentIds>> retrieveAllMessageIdAttachmentIds() {
+        return cassandraAsyncExecutor.execute(
+            selectAllMessagesWithAttachment.bind()
+                .setReadTimeoutMillis(configuration.getMessageAttachmentIdsReadTimeout()))
```

Comment:
I don't understand why we use such a huge timeout. Cassandra will do multiple fetches based on fetchSize param (5000 by default). Does the timeout applies to the whole query or only a single fetch ?

Reply:
Whole query for my experience.

Reply:
Looking at the driver code, I doubt that's the case. I propose to keep that configuration just in case but to a small value.
Maybe the driver upgrade changed that, it would be great to know more.

Reply:
changed to 10mn, we'll see